python/cuml/solvers/cd_mg.pyx

```diff
@@ -0,0 +1,132 @@
+#
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+import ctypes
+import cudf
+import numpy as np
+import rmm
+
+from libcpp cimport bool
+from libc.stdint cimport uintptr_t, uint32_t, uint64_t
+from cython.operator cimport dereference as deref
+
+from cuml.common.base import Base
+from cuml.common.array import CumlArray
+from cuml.common.handle cimport cumlHandle
+from cuml.common.opg_data_utils_mg cimport *
+from cuml.common.input_utils import input_to_cuml_array
+from cuml.decomposition.utils cimport *
+from cuml.linear_model.base_mg import MGFitMixin
+from cuml.solvers import CD
+
+cdef extern from "cumlprims/opg/cd.hpp" namespace "ML::CD::opg":
+
+    cdef void fit(cumlHandle& handle,
+                  RankSizePair **rank_sizes,
+                  size_t n_parts,
+                  floatData_t **input,
+                  size_t n_rows,
+                  size_t n_cols,
+                  floatData_t **labels,
+                  float *coef,
+                  float *intercept,
+                  bool fit_intercept,
+                  bool normalize,
+                  int epochs,
+                  float alpha,
+                  float l1_ratio,
+                  bool shuffle,
+                  float tol,
+                  bool verbose) except +
+
+    cdef void fit(cumlHandle& handle,
+                  RankSizePair **rank_sizes,
+                  size_t n_parts,
+                  doubleData_t **input,
+                  size_t n_rows,
+                  size_t n_cols,
+                  doubleData_t **labels,
+                  double *coef,
+                  double *intercept,
+                  bool fit_intercept,
+                  bool normalize,
+                  int epochs,
+                  double alpha,
+                  double l1_ratio,
+                  bool shuffle,
+                  double tol,
+                  bool verbose) except +
+
+
+class CDMG(MGFitMixin, CD):
```

Comment:
i think a few lines of documentation about the class will be helpful

Reply:
We don't do that for MG models currently because they are not meant for user consumption, this models are now simple and narrow focused that I think they are self explanatory